### PR TITLE
Normalize paths, also split drive and path on windows

### PR DESCRIPTION
**Pull request by gaborcsardi**
It is probably super slow, but at least correct. 

Btw. we could think about how to handle the win/posix cases in general. It would be great to be able to handle win paths on posix as well and vica versa. Anyway, I'll open an issue for that later.

### CODE REVIEW

**Comment by gaborcsardi** [comment]
Will fix the Travis error in a sec.

**Comment by gaborcsardi** [comment]
Should be good now. I might need to update it, depending on whether you merge this or #8 first.

**Comment by richfitz** [comment]
I thought I had merged #8 already but obviously not :koala:

I should really just give you push access.  And definitely your next PR better must you added as an author.  Let me look over all of this tomorrow and I'll get it all straightened out.

**Comment by gaborcsardi** [comment]
It's all good, looks like you can actually merge it. Makes sense, git just knows that part of this was merged already....

**Comment by richfitz** [comment]
> Btw. we could think about how to handle the win/posix cases in general. It would be great to be able to handle win paths on posix as well and vica versa. Anyway, I'll open an issue for that later.

I was thinking we could implement both sets of functions, and then use one of the ways that R allows platform dependence to set things up to behave appropriately at package build or load.  But there are other options such as posix everywhere by default given that R/msys/etc basically runs in a fake unix system anyway.

**Comment by richfitz** [comment]
Is the Collate (via `@include`) really needed here?  I really am not a fan of it and unless I am missing something I don't think it's needed here...

**Comment by gaborcsardi** [comment]
YEs, you are right, `Collate` is not needed.

File: R/normalize.R [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+path_split_drive <- function(path) {
+  lapply(path, path_split_drive1)
+}
+
+path_split_drive1 <- function(path1) {
+  npath1 <- gsub("/", "\\", path1, fixed = TRUE)
+  if (startswith(npath1, "\\\\") && substr(npath1, 3, 3) != "\\") {
+    if (grepl("^\\\\\\\\[^\\\\]+\\\\[^\\\\]+.*$", npath1)) {
+      nunc <- sub("^(\\\\\\\\[^\\\\]+\\\\[^\\\\]+).*$", "\\1", npath1)
```
Comment (richfitz) [inline R/normalize.R:162]: I believe this is the backslash to end all other text, but short of the true name of ba'al, the soul-eater https://xkcd.com/1638/

**Comment by richfitz** [comment]
We'll clip it out later to keep the merges simple.

File: R/normalize.R (hunk 1 above)
Comment (gaborcsardi) [inline R/normalize.R:162]: LOL.

**Pull request merged**